### PR TITLE
Liquidity mining

### PR DESCRIPTION
**Pull request by ororopickpocket**

### CODE REVIEW

File: rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol [hunk 1]
```diff
@@ -139,23 +171,29 @@ contract RBTCWrapperProxy is ContractRegistryClient {
...
+        ISmartToken _poolToken = _liquidityPoolConverter.poolToken(reserveToken);
 
-        IWrbtcERC20(wrbtcTokenAddress).deposit.value(_amount)();
+        //wrap rbtc if required
+        if(_reserveAddress == wrbtcTokenAddress){
+            require(_amount == msg.value, "The provided amount should be identical to msg.value");
+            IWrbtcERC20(wrbtcTokenAddress).deposit.value(_amount)();
+        }
+        else{
+            reserveToken.transferFrom(msg.sender, address(this), _amount);
```
Comment (jameshowlett977) [inline rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol:184]: Please add `require`

**Review by jameshowlett977 — APPROVED** [review]

File: rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol [hunk 2]
```diff
@@ -124,13 +153,15 @@ contract RBTCWrapperProxy is ContractRegistryClient {
      * 4.Transfers pool tokens to user
```
Comment (eMarchenko) [inline rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol:153]: the function sends pool tokens to `liquidityMiningContract` instead

same for `addLiquidityToV1`

File: rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol [hunk 3]
```diff
@@ -216,8 +253,9 @@ contract RBTCWrapperProxy is ContractRegistryClient {
         _liquidityPoolConverter.addLiquidity(_reserveTokens, _reserveAmounts, _minReturn);
         uint256 poolTokenAmount = _poolToken.balanceOf(address(this)).sub(poolTokenAmountBefore);
         
-        success = _poolToken.transfer(msg.sender, poolTokenAmount);
-        require(success, "Failed to transfer pool token to user");
+        //deposit the pool tokens in the liquidity mining contract on the sender's behalf
+        _poolToken.approve(address(liquidityMiningContract), poolTokenAmount);
```
Comment (eMarchenko) [inline rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol:257]: formally must check return value of `approve`

File: rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol [hunk 4]
```diff
@@ -251,13 +289,15 @@ contract RBTCWrapperProxy is ContractRegistryClient {
      * 4.Sneds RBTC to user
```
Comment (eMarchenko) [inline rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol:289]: typo - `sends`
you also might want to update process description

same for `removeLiquidityFromV1`

**Review by eMarchenko — APPROVED** [review]
minor issues, that don't affect security

File: rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol [hunk 5]
```diff
@@ -19,6 +20,7 @@ contract RBTCWrapperProxy is ContractRegistryClient {
 
     address public wrbtcTokenAddress;
     address public sovrynSwapNetworkAddress;
```
Comment (eMarchenko) [inline rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol:22]: there is `ContractRegistryClient.addressOf(SOVRYNSWAP_NETWORK)`, shouldn't it be used intead?

**Review by eMarchenko — APPROVED** [review]
minor issues, that don't affect security

**Pull request merged**